Sources/MongoSwift/BSON/BSONValue.swift

```diff
@@ -542,21 +720,39 @@ extension Int: BSONValue {
         throw RuntimeError.internalError(message: "`Int` value \(self) could not be encoded as `Int32` or `Int64`")
     }
 
+    public func bsonEquals(_ other: BSONValue?) -> Bool {
```

Comment:
we discussed this friday but I can't quite remember, can you clarify what the behavior is for `bsonEquals` with the integer types / why we separately defining `bsonEquals` for them?

Reply:
We need separate implementations of `bsonEquals` for `Int`, `Int32`, and `Int64` because they need to compare against both their own type and `Int`, depending on if `Int` represents a BSON int32 or a BSON int64.

e.g. on 64 bit:
```swift
5.bsonEquals(Int64(5)) // true
5.bsonEquals(5) // true
5.bsonEquals(Int32(5)) // false

Int64(5).bsonEquals(5) // true
Int64(5).bsonEquals(Int64) // true
```

So friday I mentioned that I wanted to make some slight changes to how `bsonEquals` works, but having thought about it more, I don't think they're necessary right now (as @mbroadst already determined)